### ISSUE TITLE
Support for proxy

### ISSUE THREAD

**Issue by tribou** [Hackafe/atom-sync-settings#26]
This plugin is doing nothing, and I'm thinking it's because I'm behind a proxy.

Any ideas for supporting proxies?

e.g.
```bash
export HTTP_PROXY=proxy
export HTTPS_PROXY=proxy
```

**Comment by mak42** [comment]
Push!!!
I badly need proxy support, the node-github module should support it afaik. 
So I asume the problem is in atom-sync-settings?!?

**Comment by jcare44** [comment]
Same problem for me on windows 7 with Atom 1.0.
Here is my stack trace :
```
SyntaxError: Unexpected token s
  at Object.parse (native)
  at C:\Users\juca\.atom\packages\sync-settings\lib\sync-settings.coffee:70:26
  at Object.proto.sendError (C:\Users\juca\.atom\packages\sync-settings\node_modules\github\api\v3.0.0\index.js:32:13)
  at C:\Users\juca\.atom\packages\sync-settings\node_modules\github\api\v3.0.0\gists.js:180:29
  at callCallback (C:\Users\juca\.atom\packages\sync-settings\node_modules\github\index.js:743:17)
  at ClientRequest.<anonymous> (C:\Users\juca\.atom\packages\sync-settings\node_modules\github\index.js:809:17)
  at emitOne (events.js:77:13)
  at ClientRequest.emit (events.js:166:7)
  at TLSSocket.socketErrorListener (_http_client.js:254:9)
  at emitOne (events.js:77:13)
  at TLSSocket.emit (events.js:166:7)
  at TLSSocket.onHangUp (_tls_wrap.js:953:14)
  at TLSSocket.g (events.js:257:16)
  at emitNone (events.js:72:20)
  at TLSSocket.emit (events.js:163:7)
  at _stream_readable.js:891:16
  at process._tickCallback (node.js:357:13)
```

**Comment by asolfre** [comment]
Seems related to #51

**Comment by SaintPeter** [comment]
I, too, need proxy support.  I was able to install on my corporate PC after setting up the proxy variable on NPM, but it looks like sync-settings needs it as well.

**Comment by jgopel** [comment]
What do you need as far as repro? I can consistently get it to happen on my work PC.

**Comment by DSigmund** [comment]
+1 Need proxy support

**Comment by Mihai-P** [comment]
I am behind a proxy too and I get this error. I have a proxy set up in atom.

**Comment by SaintPeter** [comment]
FYI, I dug into this a bit.

First, the node-github library that this plugin uses does have some support for proxy as a config value and from an :
https://github.com/mikedeboer/node-github/blob/master/index.js#L673

Here is the relevant code:
```js
var proxyUrl;
        if (this.config.proxy !== undefined) {
            proxyUrl = this.config.proxy;
        } else {
            proxyUrl = process.env.HTTPS_PROXY || process.env.HTTP_PROXY;
        }
```

So, theoretically, this plugin should work if you set an environment variable with either`HTTP_PROXY` or `HTTPS_PROXY`.  I experimented a bit and was unable to get it to work, though.  I don't know if that's due to my local environment, or if node-github simply doesn't work properly.  If it's the latter, we can file a bug there and see if we can get it fixed.

Can anyone else on this thread give it a try and report your results?

**Comment by jerone** [comment]
On Atom Discuss there's also an topic about proxy for the interested: https://discuss.atom.io/t/is-there-any-proxy-settings/710

**Comment by Mihai-P** [comment]
I already have a proxy set up properly that works in atom. I can install packages but I believe we are blocked access to https on github so there is probably nothing around that. I did manage to sync just by using another connection without a proxy. Good job on the package. I will try in the following days to debug it to see if it uses the proxy defined for atom, I will report back afterwards.

**Comment by jmdacruz** [comment]
Let me just add that I'm also seeing this issue when using a locally installed transparent proxy such as Proxifier

**Comment by danpantry** [comment]
:+1 also experiencing this, have atom's http-proxy/https-proxy variable set up but still experiencing issues.

**Comment by zchwyng** [comment]
+1

**Comment by dirk-thomas** [comment]
This is a duplicate of #142. The feature has been implemented in #167.
